packages/react-native/sdks/hermes-engine/hermes-utils.rb

```diff
@@ -195,6 +207,60 @@ def podspec_source_download_prebuilt_nightly_tarball(version)
     return {:http => url}
 end
 
+# [macOS
+# Hermes and JSI don't always guarantee backwards compatibility, so the safest option is to use
+# the commit hash of Hermes at the time of the merge base between us and RNCore.
+def podspec_source_build_from_github_merge_base()
+    rncore_main = `git ls-remote https://github.com/facebook/react-native.git main | cut -f 1`.strip
+    if rncore_main.empty?
+        abort <<-EOS
+        [Hermes] Unable to find the main branch commit hash of RNCore.
+        EOS
+    end
+
+    fetch_result = `git fetch -q https://github.com/facebook/react-native.git #{rncore_main}`
+    if $?.exitstatus != 0
+        abort <<-EOS
+        [Hermes] Failed to fetch RNCore main commit (#{rncore_main}) into the local repository.
+        EOS
+    end
+
+    merge_base = `git merge-base #{rncore_main} HEAD`.strip
+    if merge_base.empty?
+        abort <<-EOS
+        [Hermes] Unable to find the merge base between RNCore main (#{rncore_main}) and the current branch.
+        EOS
+    end
+
+    timestamp = `git show -s --format=%ci #{merge_base}`.strip
+    if timestamp.empty?
+        abort <<-EOS
+        [Hermes] Unable to extract the timestamp for the merge base (#{merge_base}).
+        EOS
+    end
+
+    commit = nil
+    Dir.mktmpdir do |tmpdir|
+        hermes_git_dir = File.join(tmpdir, "hermes.git")
+        # Unfortunately we can't use git rev-list on HERMES_GITHUB_URL directly since we're not in that repo.
+        # Instead, we create a shallow clone to avoid downloading the entire history.
+        `git clone -q --bare --shallow-since="#{timestamp}" #{HERMES_GITHUB_URL} "#{hermes_git_dir}"`
+        `git --git-dir="#{hermes_git_dir}" fetch -q --deepen=1`
+
+        # If all goes well, this will be the commit hash of hermes at the time of the merge base
+        commit = `git --git-dir="#{hermes_git_dir}" rev-list -1 --before="#{timestamp}" HEAD`.strip
```

Comment:
Uh, it feels pretty heavy that now running "pod install" makes a guy clone of rn in /tmp 😅

Reply:
I'm also not a big fan of this. Since we're always chasing upstream, can we snap to whatever version they're using instead of building from source?

Reply:
I guess this is only on the main branch, where React Native Core was already calling git ls-remote. When we enable nightlies, maybe we can snap Hermes to a nightly commit and not need the merge base, but I suppose that can be a followup after we enable Hermes on the stable branches. I'll resolve this for now.